src/main/java/uk/gov/justice/hmpps/prison/service/FinanceService.java

```diff
@@ -41,12 +47,34 @@
     private final AccountCodeRepository accountCodeRepository;
     private final OffenderSubAccountRepository offenderSubAccountRepository;
     private final OffenderTrustAccountRepository offenderTrustAccountRepository;
+    private final OffenderDamageObligationService offenderDamageObligationService;
 
     @VerifyBookingAccess(overrideRoles = {"SYSTEM_USER", "GLOBAL_SEARCH"})
     public Account getBalances(final Long bookingId) {
-        return bookingRepository.getBookingAgency(bookingId)
-                .map(agency -> financeRepository.getBalances(bookingId, agency))
-                .orElse(null);
+
+        final var offenderSummary = bookingRepository.getLatestBookingByBookingId(bookingId)
+            .orElseThrow(new EntityNotFoundException("Booking not found for id: " + bookingId));
+
+        final var damageObligationBalance =
+            offenderDamageObligationService.getDamageObligations(offenderSummary.getOffenderNo(), "")
+                .stream()
+                .map(OffenderDamageObligationModel::getAmountToPay)
+                .reduce(BigDecimal.ZERO, BigDecimal::add);
+
+        return Optional.ofNullable(financeRepository.getBalances(bookingId, offenderSummary.getAgencyLocationId()))
+            .map(Account::toBuilder)
+            .map(builder -> builder.damageObligations(toMoneyScale(damageObligationBalance)))
+            .map(AccountBuilder::build)
+            .orElse(zeroBalances());
+    }
+
+    private Account zeroBalances() {
+        return Account.builder()
+            .spends(toMoney("0.00"))
+            .cash(toMoney("0.00"))
+            .savings(toMoney("0.00"))
+            .damageObligations(toMoney("0.00"))
+            .currency("£").build();
```

Comment:
I think that needs to be GBP. Best to pull it from properties - @Value("${api.currency}")

Reply:
done